Source/Xamarin/Prism.DryIoc.Forms/DryIocExtensions.cs

```diff
@@ -11,34 +12,150 @@ namespace Prism.DryIoc
+        /// <param name="tabletView">Tablet Specific View Type</param>
+        /// <param name="phoneView">Phone Specific View Type</param>
+        public static void RegisterTypeForNavigationOnIdiom<TView, TViewModel>(this IContainer container, string name = null, Type desktopView = null, Type tabletView = null, Type phoneView = null)
+            where TView : Page
+            where TViewModel : class
+        {
+            if (Device.Idiom == TargetIdiom.Desktop && desktopView != null)
+            {
+                container.RegisterTypeForNavigationWithViewModel<TViewModel>(desktopView, name);
+            }
+            else if (Device.Idiom == TargetIdiom.Phone && phoneView != null)
+            {
+                container.RegisterTypeForNavigationWithViewModel<TViewModel>(phoneView, name);
+            }
+            else if (Device.Idiom == TargetIdiom.Tablet && tabletView != null)
+            {
+                container.RegisterTypeForNavigationWithViewModel<TViewModel>(tabletView, name);
+            }
+            else
+            {
+                container.RegisterTypeForNavigation<TView, TViewModel>(name);
+            }
+        }
+
+        //TODO: decide if Prism will continue to support navigating via ViewModels, or should we stick to just a single approach to navigation
+        //public static void RegisterTypeForViewModelNavigation<TView, TViewModel>(this IContainer container)
+        //    where TView : Page
+        //    where TViewModel : class
+        //{
+        //    container.RegisterTypeForNavigationWithViewModel<TViewModel>(typeof(TView), typeof(TViewModel).FullName);
+        //}
+
+        //public static void RegisterTypeForViewModelNavigationOnPlatform<TView, TViewModel>(this IContainer container, Type AndroidView = null, Type iOSView = null, Type OtherView = null, Type WindowsView = null, Type WinPhoneView = null)
+        //    where TView : Page
+        //    where TViewModel : class
+        //{
+        //    container.RegisterTypeForNavigationOnPlatform<TView, TViewModel>(typeof(TViewModel).FullName, AndroidView, iOSView, OtherView, WindowsView, WinPhoneView);
+        //}
+
+        //public static void RegisterTypeForViewModelNavigationOnIdiom<TView, TViewModel>(this IContainer container, Type DesktopView = null, Type TabletView = null, Type PhoneView = null)
+        //    where TView : Page
+        //    where TViewModel : class
+        //{
+        //    container.RegisterTypeForNavigationOnIdiom<TView, TViewModel>(typeof(TViewModel).FullName, DesktopView, TabletView, PhoneView);
+        //}
+
+        private static void RegisterTypeForNavigationWithViewModel<TViewModel>(this IContainer container, Type viewType, string name = null)
+            where TViewModel : class
         {
-            container.RegisterTypeForNavigation<TView>(typeof(TViewModel).FullName);
+            if (string.IsNullOrWhiteSpace(name))
+                name = viewType.Name;
```

Comment:
Great catch!  I actually meant to do another name check using typeof(TView).Name to use as the name in this method.  This keeps the expectation of the naming convention consistent with the other registration methods.  I'll fix that.